test/lint/lint-cppcheck-dash.sh

```diff
@@ -0,0 +1,91 @@
+#    "Member variable '.*' is not initialized in the constructor."
+#    "Consider using std::copy algorithm instead of a raw loop."
+#    "Consider using std::any_of algorithm instead of a raw loop."
+#    "Consider using std::accumulate algorithm instead of a raw loop."
+#    "Consider using std::copy algorithm instead of a raw loop."
+#    "Consider using std::transform algorithm instead of a raw loop."
+#    "Consider using std::find_if algorithm instead of a raw loop."
+)
+
+IGNORED_WARNINGS=(
+    "src/bls/bls.h:.* warning: Struct 'CBLSIdImplicit' has a constructor with 1 argument that is not explicit."
+    "src/llmq/quorums_init.cpp:.* warning: Variable 'llmqDbTmp' is assigned a value that is never used."
+    "src/llmq/quorums.cpp:.* warning: Parameter '_blsWorker' can be declared with const"
+
+# General catchall, for some reason any value named 'hash' is viewed as never used.
+    "Variable 'hash' is assigned a value that is never used."
+
+# The following can be useful to ignore when the catch all is used
+#    "Consider performing initialization in initialization list."
+#    "Consider using std::transform algorithm instead of a raw loop."
+#    "Consider using std::accumulate algorithm instead of a raw loop."
+#    "Consider using std::any_of algorithm instead of a raw loop."
+#    "Consider using std::count_if algorithm instead of a raw loop."
+#    "Member variable '.*' is not initialized in the constructor."
+#    "Consider using std::find_if algorithm instead of a raw loop."
+)
+
+# We should attempt to update this with all dash specific code
+FILES=$(git ls-files -- "src/bls/*.cpp" "src/bls/*.h" "src/coinjoin/*.cpp" "src/coinjoin/*.h" "src/evo/*.cpp" \
+                        "src/evo/*.h" "src/governance/*.cpp" "src/governance/*.h" "src/llmq/*.cpp" "src/llmq/*.h" \
+                        "src/masternode/*.cpp" "src/masternode/*.h" "src/dsnotificationinterface.*" "src/spork.*" \
+                        "src/rpc/coinjoin.cpp" "src/rpc/governance.cpp" "src/rpc/masternode.cpp" "src/rpc/rpcevo.cpp" \
+                        "src/rpc/rpcquorums.cpp")
+
+
+if ! command -v cppcheck > /dev/null; then
+    echo "Skipping cppcheck linting since cppcheck is not installed. Install by running \"apt install cppcheck\""
+    exit 0
+fi
+
+function join_array {
+    local IFS="$1"
+    shift
+    echo "$*"
+}
+
+ENABLED_CHECKS_REGEXP=$(join_array "|" "${ENABLED_CHECKS[@]}")
+IGNORED_WARNINGS_REGEXP=$(join_array "|" "${IGNORED_WARNINGS[@]}")
+FILES_REGEXP=$(join_array "|" "${FILES[@]}")
+WARNINGS=$(echo "${FILES}" | \
+    xargs cppcheck --enable=all -j "$(getconf _NPROCESSORS_ONLN)" --language=c++ --std=c++17 --template=gcc -D__cplusplus -DENABLE_WALLET -DCLIENT_VERSION_BUILD -DCLIENT_VERSION_IS_RELEASE -DCLIENT_VERSION_MAJOR -DCLIENT_VERSION_MINOR -DCLIENT_VERSION_REVISION -DCOPYRIGHT_YEAR -DDEBUG -DHAVE_WORKING_BOOST_SLEEP_FOR -I src/ -q 2>&1 | sort -u | \
```

Comment:
c++14 is the required version, c++17 is an optional one, so I think this should be
```suggestion
    xargs cppcheck --enable=all -j "$(getconf _NPROCESSORS_ONLN)" --language=c++ --std=c++14 --template=gcc -D__cplusplus -DENABLE_WALLET -DCLIENT_VERSION_BUILD -DCLIENT_VERSION_IS_RELEASE -DCLIENT_VERSION_MAJOR -DCLIENT_VERSION_MINOR -DCLIENT_VERSION_REVISION -DCOPYRIGHT_YEAR -DDEBUG -DHAVE_WORKING_BOOST_SLEEP_FOR -I src/ -q 2>&1 | sort -u | \
```